drivers/at_cmd/at_cmd.c

```diff
@@ -110,7 +111,7 @@ static void callback_worker(struct k_work *item)
 	struct callback_work_item *data =
 		CONTAINER_OF(item, struct callback_work_item, work);
 
-	if ((data != NULL) && (data->callback != NULL)) {
```

Comment:
I think it is safer to keep this check.

Reply:
`data` is still checked, and `data->callback` is not supposed to be NULL after these changes, so we're safe here. We could optionally assert on `data->callback`, but I think we're good to go for now.